Document --architecture option for dotnet-sos (#22108)
dotnet/docs: docs/core/diagnostics/dotnet-sos.md

### Synopsis

```console
dotnet-sos install [--architecture <arch>]
```

### Options

- **`--architecture <arch>`**

>   Specifies the processor architecture of the SOS binaries to install. By default, `dotnet-sos` installs the architecture of the host machine. Use this option when you are trying to install SOS for a different from the dotnet host architecture - for example, if you are running Arm32 binaries from an Arm64 host, you will need to install SOS with `dotnet-sos install --architecture Arm`.

Reviewer: ```suggestion
  Specifies the processor architecture of the SOS binaries to install. By default, `dotnet-sos` installs the architecture of the host machine. Use this option when you want to install SOS for an architecture that's different from the dotnet host architecture. For example, if you're running Arm32 binaries from an Arm64 host, you will need to install SOS with `dotnet-sos install --architecture Arm`.
```

>   Specifies the processor architecture of the SOS binaries to install. By default, `dotnet-sos` installs the architecture of the host machine. Use this option when you are trying to install SOS for a different architecture from your host architecture - for example, if you are running Arm32 binaries from an Arm64 host, you will need to install SOS with `dotnet-sos install --architecture Arm`.

Reviewer 2: nit: "different architecture form your host architecture" -> "different from the dotnet host architecture"
